projectmainpage.html

```diff
@@ -0,0 +1,265 @@
+                    <li><input type="checkbox" id="nail-polish" name="nail_polish"> Nail Polish</li>
+                    <li><input type="checkbox" id="blush" name="blush"> Blush</li>
+                    <li><input type="checkbox" id="powder" name="powder"> Powder</li>
+                </ul>
+                <br>
+                <h3>Skin Tone Filters:</h3>
+                <ul>
+                    **This filter can ONLY be applied when foundation, concealer, and bronzer products are filtered**
+                </ul>
+                    <ul>
+                        <li><input type="radio" id="skin-tone-fair" name="skin-tone" value="fair"> Light</li>
+                        <li><input type="radio" id="skin-tone-medium" name="skin-tone" value="medium"> Medium</li>
+                        <li><input type="radio" id="skin-tone-dark" name="skin-tone" value="dark"> Dark</li>
+                        <li><input type="radio" id="skin-tone-none" name="skin-tone" value="none" checked> No Filter Applied</li>
+                    </ul>
+                <br>
+                <h3>Price Filters:</h3>
+                <ul>
+                    <li><input type="checkbox" id="under-10" name="under-10"> Under $10</li>
+                    <li><input type="checkbox" id="10-20" name="10-20"> $10 - $20</li>
+                    <li><input type="checkbox" id="20-50" name="20-50"> $20 - $50</li>
+                    <li><input type="checkbox" id="50-plus" name="50-plus"> $50 and above</li>
+                </ul>
+            </aside>
+
+            <!-- recommendations section with headers-->
+            <section id="home-recommendations">
+                <h2>Welcome to Makeup Manager!</h2>
+                <br>
+                <br>
+                <h2>Makeup reccomendations for you: </h2>
+                <br>
+                <br>
+                <div id="recommendation-grid"></div>
+            </section>
+        </main>
+
+        <br>
+        <!-- Footer -->
+        <footer id="footer">
+            <p>Copyright by MakeupManager</p>
+        </footer>
+
+        <script>
+            const recommendationsGrid = document.getElementById('recommendation-grid');
+            const filterContainer = document.getElementById('filter-container');
+            const searchBar = document.getElementById('search-bar');
+            const loadingPopup = document.getElementById('loading-popup');
+
+            let allProducts = [];
+            let filteredProducts = [];
+            let savedProducts = []; 
+
+            function showLoading() {
+                loadingPopup.style.display = 'flex';
+                loadingPopup.innerText = 'Loading recommendations, please wait...';
+            }
+
+            function hideLoading() {
+                loadingPopup.style.display = 'none';
+            }
+
+            showLoading();
+            fetch('http://makeup-api.herokuapp.com/api/v1/products.json')
```

Comment:
This code shouldn't be blocking! See my comment in the map function for further details. Also,

Our API has parameters!!! My point here is that we shouldn't be downloading the 1.6MB file to every client since the API rate-limits 1615.287kB / 7.1s = ~228kBps (source: Dev Tools Network tab with cache disabled).

Parameter | Data Type | Description
-- | -- | --
product_type | string | The type of makeup being searched for (ie. lipstick, eyeliner). See list of product types below. Will return a list of all products of this type
product_category | string | Sub-category for each makeup-type. (ie. lip gloss is a category of lipstick). See product types below. If a category exists it will be under 'By Category'. Will return a list of all products of this category
product_tags | string, list separated by commas | Options each product could be tagged with. (ie. vegan). each product can have multiple tags. If tags exist it will be in the product types under 'By Tag'. Will return a list of products filtered by all tags indicated
brand | string | Brand of the product. Will return all products for each brand
price_greater_than | number | Will return a list of products with price greater than indicated number (exclusive)
price_less_than | number | Will return a list of products with price less than indicated number (exclusive)
rating_greater_than | number | Will return a list of products with a rating more than indicated number (exclusive)
rating_less_than | number | Will return a list of products with a rating less than indicated number (exclusive)

Reply:
Fixed.

NOTE: This is a placeholder anyway?